### PR TITLE
API-75: Get and list channels

### PR DESCRIPTION
**Pull request by fitn**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**
This PR is related to this feature: https://akeneo.atlassian.net/browse/API-75
It allows to list and get a channel through the API

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | ?
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | OK
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/Controller/ChannelController.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        }
+
+        $channelStandard = $this->normalizer->normalize($channel, 'external_api');
+
+        return new JsonResponse($channelStandard);
+    }
+
+    /**
+     * @param Request $request
+     *
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/Controller/ChannelController.php]: @throws UnprocessableEntityHttpException

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Channel/ListChannelIntegration.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            ]
+        ];
+
+        $response = $client->getResponse();
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
+
+        $this->assertSame($standardChannels, json_decode($response->getContent(), true));
+    }
+
+    public function testOutOfRangeListChannels()
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Channel/ListChannelIntegration.php]: can you add some test with limit different to 10 ?

File: src/Pim/Bundle/ApiBundle/Controller/ChannelController.php (hunk 1 above)
Comment (fitn) [inline src/Pim/Bundle/ApiBundle/Controller/ChannelController.php]: Done

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Channel/ListChannelIntegration.php (hunk 2 above)
Comment (fitn) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Channel/ListChannelIntegration.php]: Done

File: src/Pim/Bundle/ApiBundle/Controller/ChannelController.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            throw new UnprocessableEntityHttpException($e->getMessage(), $e);
+        }
+
+        $offset = $queryParameters['limit'] * ($queryParameters['page'] - 1);
+
+        $count = $this->repository->countAll();
+
+        $channels = $this->repository->findBy([], ['code' => 'ASC'], $queryParameters['limit'], $offset);
+
+        $channelsStandard = $this->normalizer->normalize($channels, 'external_api');
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/Controller/ChannelController.php]: not std if external api ;)

Comment (fitn) [inline src/Pim/Bundle/ApiBundle/Controller/ChannelController.php]: In fact, at this time, the standard format is returned. What would you suggest?

**Pull request merged**